src/integrations/rapier/fluids_pipeline.rs

```diff
@@ -152,108 +153,110 @@ impl<'a> CouplingManager for ColliderCouplingManager<'a> {
+                                                        normal.dot(&fluid.velocities[*particle_id]);
 
-                                                if vel_err > na::zero::<f32>() {
-                                                    fluid.velocities[*particle_id] -=
-                                                        *normal * vel_err;
+                                                    if vel_err > na::zero::<f32>() {
+                                                        fluid.velocities[*particle_id] -=
+                                                            *normal * vel_err;
+                                                    }
+                                                } else if depth > h + prediction {
+                                                    continue;
                                                 }
-                                            } else if depth > h + prediction {
-                                                continue;
                                             }
-                                        }
 
-                                        let velocity =
-                                            body.map(|b| b.velocity_at_point(&proj.point));
+                                            let velocity =
+                                                body.map(|b| b.velocity_at_point(&proj.point));
 
-                                        boundary
-                                            .velocities
-                                            .push(velocity.unwrap_or(Vector::zeros()));
-                                        boundary.positions.push(proj.point);
-                                        boundary.volumes.push(na::zero::<f32>());
-                                        coupling.features.push(feature);
+                                            boundary
+                                                .velocities
+                                                .push(velocity.unwrap_or(Vector::zeros()));
+                                            boundary.positions.push(proj.point);
+                                            boundary.volumes.push(na::zero::<f32>());
+                                            coupling.features.push(feature);
+                                        }
+                                    }
+                                    HGridEntry::BoundaryParticle(..) => {
+                                        // Not yet implemented.
                                     }
-                                }
-                                HGridEntry::BoundaryParticle(..) => {
-                                    // Not yet implemented.
                                 }
                             }
                         }
                     }
-                }
 
-                boundary.clear_forces(true);
+                    boundary.clear_forces(true);
+                }
```

Comment:
This is probably the spurious bracket that needs to be removed?

Reply:
I rebased this commit, and fixed these nits.